libs/utils/include/utils/Allocator.h

```diff
@@ -51,7 +51,7 @@ static inline P* align(P* p, size_t alignment) noexcept {
 template <typename P>
 static inline P* align(P* p, size_t alignment, size_t offset) noexcept {
     P* const r = align(add(p, offset), alignment);
-    assert(pointermath::add(r, -offset) >= p);
+    assert(r >= add(p, offset));
```

Comment:
What was the issue here?

Reply:
`offset` is an unsigned integer type and compiler warning 4146 kicked in. Our initial solution was to pragma push disable the warning, but then we figured that the inequality could be rearranged.

Do you see a specific hazard in this change? I only considered integer overflow/underflow but the diff doesn't seem to worsen the chance of that. In fact if `add(p, offset)` overflowed, `align` would break (regardless of our changes).

Reply:
sounds good. thanks.